ddtrace/contrib/kombu/patch.py

```diff
@@ -0,0 +1,91 @@
+# 3p
+import kombu
+import wrapt
+
+# project
+from ...pin import Pin
+from ...ext import kombu as kombux
+from ...utils.wrappers import unwrap
+from ...propagation.http import HTTPPropagator
+from .util import (
+    get_exchange_from_args,
+    get_body_length_from_args,
+    get_routing_key_from_args,
+    _extract_conn_tags
+)
+
+HEADER_POS = 4
+propagator = HTTPPropagator()
+
+
+def patch():
+    """Patch the instrumented methods
+
+    This duplicated doesn't look nice. The nicer alternative is to use an ObjectProxy on top
+    of Kombu. However, it means that any "import kombu.Connection" won't be instrumented.
+    """
+    if getattr(kombu, '_datadog_patch', False):
+        return
+    setattr(kombu, '_datadog_patch', True)
+
+    _w = wrapt.wrap_function_wrapper
+    _w('kombu', 'Producer._publish', traced_publish)
```

Comment:
Patching an internal method may cause issues even with bugfix releases (i.e. method changes the name or the behavior). We must be sure our system doesn't crash users applications in case the internal method changes.

Also do we have other options to instrument that library instead of monkey-patching the main entrypoints?

Reply:
I did this because it was quick/dirty/easier than mashing up the kwargs and args to find the desired fields, I'll clean this up, it was just to get *something* working.